time_heap.js

```diff
@@ -81,6 +83,17 @@ function TimeHeap(options) {
     self.end = 0;
     self.lastRun = 0;
     self.drainUntil = 0;
+
+    self.scheduledDrain = null;
+
+    self.boundDrainExpired = boundDrainExpired;
+
+    function boundDrainExpired() {
+        var now = self.scheduledDrain;
+        self.scheduledDrain = null;
+
+        self.drainExpired(now);
```

Comment:
because of the order here, it's possible for `self.drainExpired` to schedule another one... is that intended?